homeassistant/helpers/translation.py

```diff
@@ -251,6 +252,7 @@ def _build_category_cache(
         translation_strings: dict[str, dict[str, Any]],
     ) -> None:
         """Extract resources into the cache."""
+        resource: dict[str, Any] | str
```

Comment:
Shouldn't this be able to be referenced from `translation_strings`? And so we should update the type of the `translation-strings` param instead?

Reply:
The type for `translation_strings` is correct as far as I can tell. `Any` is basically `dict[str, Any] | str` there.

> Shouldn't this be able to be referenced from translation_strings?

Yeah it is, but that's also the issue. Just from `translation_strings`, `resource` would be typed as `dict[str, Any]` and thus wouldn't allow `str`. The alternative would be to rename `resource` in the third for loop to something like `resource2`. That would allow for dedicated types.

Otherwise, we need the type definition here. The type inference still works like you would expect it to. I.e. inside the first the `for ... translation_strings.values()` block, `resource` is only `dict[str, Any]`.